### PR TITLE
Switch frontend to use Docker

### PR DESCRIPTION
**Pull request by johnennewdeeson**
Hi, I'd like for frontend to use Docker for the node and php related tasks it performs. I've had a crack at breaking up the config project so that it expects PHP to be elsewhere. This means that its doing less than it was, but also means everything is more opinionated - we kind of expect this framework to be used with our d8-quickstart and with docker as we configure it.

### CODE REVIEW

**Comment by johnennewdeeson** [comment]
Probably worth having a look through the README for the Example project I was putting together. This example project can be used as a training resource for new starters who can see how it all is supposed to fit together. I like the short number of steps to get both a working Drupal site and a working frontend build up and running - https://github.com/teamdeeson/cdd-demo

**Comment by johnennewdeeson** [comment]
@notmodified can you approve this?!

File: bin/router.php [hunk 1]
```diff
@@ -27,8 +27,22 @@
...
+
     http_response_code(404);
     print '<h1>404 Not Found</h1>';
+    print "<ul>";
+    print "<li>Could not find {$docroot}{$page}</li>";
+    print "<li>Docroot: {$docroot}</li>";
+    print "<li>Page: {$page}</li>";
+    print "</ul>";
+    print "<pre>";
+    var_dump($_ENV);
```
Comment (markpdeeson) [inline bin/router.php]: Is this safe? I know this _probably_ only runs locally, but $_ENV can contain sensitive information. Do we guarantee this doesn't run on production systems?

File: bin/router.php [hunk 2]
```diff
@@ -15,7 +15,7 @@
     // Issue a redirect rather than rewrite because other servers won't assume
     // index.twig.html is an index file and we don't want to encourage
     // <a href="/"… in page files.
-    $indexes = ['/index.twig.html', '/index.php.html'];
+    $indexes = ['index.php', '/index.twig.html', '/index.php.html'];
```
[the remark this answers is not in the document]
Comment (johnennewdeeson) [reply bin/router.php]: um... @adam-dg did it work?

File: bin/router.php (hunk 1 above)
Comment (johnennewdeeson) [reply bin/router.php]: I've kind of forgotten how it all works... @adam-dg / @notmodified - what's your thoughts...

Comment (adam-dg) [reply bin/router.php]: I don't know that $_ENV will contain anything I'd consider sensitive but this script will still be present on production systems deep down inside the node_modules directory with in our projects.

Now, the way we organise our projects means that the node_modules directory is outside of the document root on the platforms that we use so this file should never be served.

File: bin/router.php (hunk 2 above)
[the remark this answers is not in the document]
Comment (adam-dg) [reply bin/router.php]: I'll check and confirm

[the remark this answers is not in the document]
Comment (adam-dg) [reply bin/router.php]: As far as I can tell it makes no difference if any of the elements of the $indexes array have leading forward slashes or not, index.php included.

**Comment by adam-dg** [comment]
This has been replaced by this https://github.com/teamdeeson/deeson-webpack-config/pull/30

**Pull request closed without merging**